Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+/*****************************************************************************
+ * VLCRendererDiscovererManager.swift
+ *
+ * Copyright © 2018 VLC authors and VideoLAN
+ * Copyright © 2018 Videolabs
+ *
+ * Authors: Soomin Lee <bubu@mikan.io>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+class VLCRendererDiscovererManager: NSObject {
+    @objc static let sharedInstance = VLCRendererDiscovererManager(presentingViewController: nil)
```

Comment:
instead of having a sharedinstance we should use dependency injection. Also this is not how you create a sharedinstance you miss the dispatch_once but really this sharedinstance needs to go away

Reply:
I don't think we need the `dispatch_once`. 
DI, sure but wouldn't it better to wait till the coordinators are removed or we completely decide to use a coordinator pattern?

Reply:
it has nothing to do with the coordinator. dependency injection would also be used if we turn the coordinators into viewcontrollers

Reply:
Well a little bit no? Wouldn't it has to do with who instantiates the manager?

Reply:
There is a private init on the manager 😢

Reply:
I was wrong you were right 😁 regarding the private init. You've done it right. and regarding the who is initialising the manager it would be a veiwcontroller or whatever object takes the place of the coordinator